src/main/java/net/minestom/server/network/packet/client/handshake/HandshakePacket.java

```diff
@@ -52,7 +53,22 @@ public void process(@NotNull PlayerConnection connection) {
             if (address != null) {
```

Comment:
bungeeguard itself seems to reject the handshake packet if it's too big, not sure how you would do that here but i think it's designed to mitigate DoS attacks?